src/bsoncxx/document/value.hpp

```diff
@@ -100,6 +119,21 @@ class BSONCXX_API value {
     ///
     BSONCXX_INLINE operator document::view() const noexcept;
 
+    ///
+    /// Constructs an object of type T from this document object. This method uses
+    /// argument-dependent lookup to find the function declaration
+    /// `void from_bson(T& t, const bsoncxx::document::view& doc)`.
+    ///
+    /// @param t
+    ///   An optional partially constructed object. If no argument is passed, the function will
+    ///   create an object with its default constructor.
+    ///
+    template <typename T>
+    T get(T t = {}) {
```

Comment:
By accepting an argument of type `T`, this lets users convert to type `T` without explicitly specifying the type. E.g. they can write:

```
Person p;
doc.get(p)
```

But, I think we want the argument to be a reference `T& t`. Otherwise, the example above does not modify `p`, so users still need to store the return value. As it stands, users need to do one of the following:
```
Person p;
p = doc.get(p);
// or
Person p;
p = doc.get<Person>();
```

I'm assuming this corresponds to the `get_to` method in nlohmann/json: https://nlohmann.github.io/json/doxygen/classnlohmann_1_1basic__json.html

Reply:
Ah, I see this was brought up in prior comment: https://github.com/mongodb/mongo-cxx-driver/pull/726#discussion_r501231115. @bazile-clyde does taking a `T&` seem right?

Reply:
Ah yes! A reference does seem better.